samples/sampler/src/ThemeEnvironment/ThemeEnvironment.js

```diff
@@ -65,23 +73,124 @@ const StorybookDecorator = (story, config = {}) => {
 		classes.debug = true;
 	}
 
+	// beginning of custom theme code
+	const request = new LS2Request();
+	const [context, setContext] = useState(customColorsContext);
+
+	const defaultKeyThemeValue = JSON.stringify({
+		"version": "0.1",
+		"activeTheme": "defaultTheme",
+		"backgroundColor": "#000000",
+		"componentBackgroundColor": "#7D848C",
+		"focusBackgroundColor": "#E6E6E6",
+		"popupBackgroundColor": "#575E66",
+		"subtitleTextColor": "#ABAEB3",
+		"textColor": "#E6E6E6",
+		colors: generateStylesheet(
+			"#000000",
+			"#7D848C",
+			"#E6E6E6",
+			"#575E66",
+			"#ABAEB3",
+			"#E6E6E6"
+		)
+	});
+
+	useEffect(() => {
+		if (platform.tv) {
+			request.send({
+				service: 'luna://com.webos.service.settings/',
+				method: 'getSystemSettings',
+				parameters: {
+					category: 'customUi',
+					keys: ['theme']
+				},
+				onSuccess: (res) => {
+					if (res.settings.theme !== '' && res) {
+						const parsedKeyData = JSON.parse(res.settings.theme);
+						setContext({...parsedKeyData});
+						// if `theme` key is an empty string, update the context with a default value, then make a SET call to service settings and set
+						// `theme` key  with a default value
+					} else if (res.settings.theme === '') {
+						setContext(JSON.parse(defaultKeyThemeValue));
+						request.send({
```

Comment:
Sampler only reads settings' values not writes. So, let's update this part.

Reply:
Okay, this part is updated to only read values